lua/telekasten.lua

```diff
@@ -216,20 +258,39 @@ local function calenderinfo_today()
         .. daysuffix(dinfo.day)
         .. ", "
         .. dinfo.year
-    opts.week = os.date("%V")
+    opts.week = os.date(dateformats.week)
+    opts.isoweek = os.date(dateformats.isoweek)
     opts.month = dinfo.month
     opts.year = dinfo.year
     opts.day = dinfo.day
     return opts
 end
 
-local function linesubst(line, title, calendar_info)
-    local cinfo = calendar_info or calenderinfo_today()
+local function linesubst(line, title, calendar_info, relative_dates)
+    local cinfo  = calendar_info or calenderinfo_today()
+    local rdates = relative_dates or relativedates_today()
```

Comment:
This is dangerous! If `calendar_info` is filled with a date, e.g. one picked from the calendar, then `calendar_info` and `rdates` will be out of sync. So I recommend (because I don't want to say request) that you derive the rdates from the `cinfo`. So instead of having a `relativedates_today()` make it a `relativedates(reference_cinfo)`. That way, rdates and dates will always be in sync.

Here is how it works:

---

Use-case: `goto_today()` is called: Here, everything is fine because the reference date is today. So the opts passed to
`GotoToday` will be nil - and hence will be filled with `calenderinfo_today()`. These opts are then passed on to
`create_note_from_template()` as last parameter, which in turn passes it on to `linesubst()` as `calendar_info`
parameter. **Note** that `GotoToday()` does **not** pass any relative dates info on to `create_note_from_template()`.

---

Use-case: `CalendarAction()` is called by picking a date on the calendar. Here, all the info the calendar plugin gives
us, is passed in: 

```lua
local function CalendarAction(day, month, year, weekday, _)
```

So in `CalendarAction`, the `opts` for `GotoToday(opts)` are constructed. We could call them `calendar_info`. When
`CalendarAction` calls `GotoToday(opts)`, the basic 'today' info is passed on, but no relative dates are worked out.
`GotoToday` passes no relative dates on to `create_note_from_template`. The parameter is omitted. Hence
`create_note_from_template` **always** passes `nil` to `linesubst` as `relative_dates` parameter which forces
`linesubst` to use `relativedates_today()`.

You can check out what happens when you put `{{yesterday}}` into your daily note template and then picking a day from the future in the calendar by pressing enter on it: the created note will have today's yesterday filled in for
`{{yesterday}}` as opposed to the day before the requested date.

---

**To make a long story short**, the best way to fix this is 

- turn `relativedates_today()` into `relativedates(cinfo)`
    - make sure to add to the passed in `cinfo` instead of separating calendar info and relative dates
- omit the never-used parameter `relative_dates` in both `create_note_from_template()` and `linesubst()`. 

That way all substitution fields in `linesubst()` will 
- always be in one `opts` (calendar_info / what ever we want to call it) table
- always be in sync with the requested date, be it 'today' or any other day.

Reply:
I just got the idea to further hacking that calendar plugin so that pressing enter on a week will call `GotoWeek()` (currently named GotoThisWeek()) - so one can create weekly notes in advance or in retrospect. This is something we cannot do now. With the suggested changes, relative dates would be handled automatically.

Reply:
Great stuff!  I'm still unpacking it all, but I wanted to see if we should change the `{{yesterday}}` and `{{tomorrow}}` wording based on these changes?  For both days and weeks I think prev and next would be better, so `{{prevday}}`, `{{nextday}}`, `{{prevweek}}`, and `{{nextweek}}`.  Does that work for you?

I was just trying to make the code fit stylistically, but let me know if you'd prefer these tokens in snake_casing or camelCasing instead of just all lowercase.

Reply:
I like `{{prevday}}` and `{{nextday}}`, etc! That doesn't only work for me, I would prefer it actually! :smile: 

I think leaving them lowercase is fine. If that doesn't work, snake_casing would be the second-best fit IMHO.